woocommerce/woocommerce
# Fix Store API response for very large price amounts

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR addresses improper formatting for very high price amounts on Store API responses





### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set up a store using Euro currency and 2 decimal points (options on `/wp-admin/admin.php?page=wc-settings`)
2. Disable taxes.
3. Add a new product with the price `9223372036854775808` and take note of the product id
4. Add that product to cart
5. Verify that on `/wp-json/wc/store/v1/cart` response JSON the `items[0].prices.price` value is correctly displayed as  `922337203685477580800`
6. Verify that on `/wp-json/wc/store/v1/products/<product id>` response JSON `prices.price` value is correctly displayed as  `922337203685477580800`

Please note that totals calculations on Cart and Checkout is not properly done. This is not a consequence of this PR and will be addressed on a different issue. Same with Cart/Checkout blocks wrong rounding for values, which is a limitation from the Dinerojs npm package.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
